### PR TITLE
Feature/958 windows setup dynamic java version

### PR DESCRIPTION
**Pull request by mmastrocinque**
### Description of the Change
Adds reading from https://download.nodecdn.net/containers/atl/launcher/json/config.json for JRE information, falls back to java 17 update 9

### Testing
I've tested breaking both the URL and json properties (i.e changing object/property names/ url. 

### Related Issues
https://github.com/ATLauncher/ATLauncher/issues/958

### CODE REVIEW

File: packaging/windows-setup/installer.iss [hunk 1]
```diff
@@ -60,8 +60,67 @@ Type: filesandordirs; Name: "{app}\jre"; Components: java
...
+        if ParseJsonAndLogErrors(JsonParser, Json) then
+          begin
+            JsonRoot := GetJsonRoot(JsonParser.Output);
+            if not FindJsonObject(JsonParser.Output, JsonRoot, 'bundledJre', BundledJreObject) or
+            not FindJsonObject(JsonParser.Output, BundledJreObject, OS, OSObject) or
+            not FindJsonString(JsonParser.Output, OSObject, 'url', Url) or
+            not FindJsonString(JsonParser.Output, OSObject, 'hash', Hash) or
+            not FindJsonString(JsonParser.Output, OSObject, 'folder',Folder) then
+            begin
+              RaiseException('Failed to read from ' + CONFIGURL + ', falling back to defaults')
```
Comment (mmastrocinque) [inline packaging/windows-setup/installer.iss:108]: This is the first time I've ever written Pascal, didn't think of a better way to handle these two error conditions but perhaps this repeated RaiseException could be simplified to one.

**Comment by RyanTheAllmighty** [comment]
Sorry on the delay on this, looks good to me, good thinking with the fallback values 👍🏻

**Comment by RyanTheAllmighty** [comment]
Hey @mmastrocinque sorry to be a pain, but are you able to raise a new PR from your forked branch into master with this. I merged develop into master, but for some reason the merge commit for this PR just got removed and no longer exists

**Comment by mmastrocinque** [comment]
Sure, I'm at work but can get to it after. I do see I have commits that were from dev that seem to be missing from master, do you want me to just fork master and re-apply my changes and not bring these other commits over? 
image

**Comment by RyanTheAllmighty** [comment]
Yeah I think that would be the best way. Still not sure why they completely disappeared :\

**Comment by RyanTheAllmighty** [comment]
@mmastrocinque managed to figure it out, thankfully had another computer with the develop branch still around, so managed to merge that into master and keep your changes as they were when they merged in https://github.com/ATLauncher/ATLauncher/commit/3bf59d51f47d95803613ced2a5a99112841cf960

Thanks again for this, will save a bunch of headcahes :)

**Comment by mmastrocinque** [comment]
> @mmastrocinque managed to figure it out, thankfully had another computer with the develop branch still around, so managed to merge that into master and keep your changes as they were when they merged in https://github.com/ATLauncher/ATLauncher/commit/3bf59d51f47d95803613ced2a5a99112841cf960
> 
> 
> 
> Thanks again for this, will save a bunch of headcahes :)

Sorry about the delay, completely forgot as I got pretty sick the last 3 days. Glad you recovered it and got it resolved!

**Pull request merged**